tests/snappi/pfc/files/helper.py

```diff
@@ -522,10 +526,35 @@ def __verify_results(rows,
                           'Total TX bytes {} should be smaller than DUT buffer size {}'.
                           format(tx_bytes_total, dut_buffer_size))
 
-            for peer_port, prios in list(flow_port_config[0].items()):
+            for peer_port, prios in flow_port_config[0].items():
                 for prio in prios:
-                    dropped_packets = get_pg_dropped_packets(
-                        duthost, peer_port, prio)
+                    dropped_packets = get_pg_dropped_packets(duthost, peer_port, prio)
                     pytest_assert(dropped_packets == 0,
                                   'Total TX dropped packets {} should be 0'.
                                   format(dropped_packets))
+
+    # Check if pause frames with zero source MAC are counted in the PFC counters
+    if pfc_pause_zero_src_mac:
+        for peer_port, prios in flow_port_config[1].items():
+            for prio in prios:
+                pfc_pause_rx_frames = get_pfc_pause_frame_count(duthost, peer_port, prio)
+                pytest_assert(pfc_pause_rx_frames > 0,
+                              "PFC pause frames with zero source MAC are not counted in the PFC counters")
+
+
+def get_pfc_pause_frame_count(duthost, port, priority):
+    """
+    Get the PFC pause frame count for a given port and priority
+    Args:
+        duthost (Ansible host instance): device under test
+        port (str): port name
+        priority (int): priority of flow
+    Returns:
+        int: PFC pause frame count
+    """
+
+    port_oid = duthost.command("sonic-db-cli COUNTERS_DB HGET COUNTERS_PORT_NAME_MAP {}".format(port))['stdout']
+    pause_frame_count = duthost.command("sonic-db-cli COUNTERS_DB HGET COUNTERS:{} SAI_PORT_STAT_PFC_{}_RX_PKTS"
+                                        .format(port_oid, priority))['stdout']
```

Comment:
please use the sonic cli to check because that is what the end user sees: "show pfc counters"

Reply:
Done